helper/resource/testing.go

```diff
@@ -576,6 +576,16 @@ type TestStep struct {
 	// otherwise an error will be returned.
 	ConfigFile config.TestStepConfigFunc
 
+	// ConfigExact indicates that the test framework should use the exact
+	// content of the Config, ConfigFile, or ConfigDirectory inputs and should
+	// not modify it at test run time.
+	//
+	// The default is false. At test run time, the test framework will generate
+	// specific kinds of configuration, such as import blocks, and append them
+	// to the given Config, ConfigFile, or ConfigDirectory inputs. Using this
+	// default improves test readability and removes duplication of setup.
+	ConfigExact bool
```

Comment:
The only qualm I have with this naming is that currently, the default for `Import` mode is that this is `false`... however if you apply this to `Config` mode, the default is technically true for `ConfigFile/Directory` and false for `Config` right?

We wouldn't be able to, in the future, make modifications based off this flag in `Config` mode because it would be a breaking change 🤔.

That's all from my memory of how we treat `Config` mode, so might need to be verified 😆, I just want to make sure we don't put ourselves in a corner

Reply:
Put differently, is there a reason we shouldn't just name this `ImportStateConfigExact` (besides how icky the name is 😆)

Reply:
> We wouldn't be able to, in the future, make modifications based off this flag in `Config` mode because it would be a breaking change 🤔.

Yes, I like this angle... design consequences 😃

Option: keep this as a global non-prefixed option, and explicitly update `Config` and `Refresh` mode to error when `true`. Then we have the future option to accept that flag in either mode without breaking tests. 

> Put differently, is there a reason we shouldn't just name this `ImportStateConfigExact` (besides how icky the name is 😆)

Also an option. I'm willing to go this route & encode the fact that we simply don't know how general this concept should be.

I'll go with that and 💭 on the naming for a moment.

Reply:
Sweet! I'm good with that 👍🏻